### ISSUE TITLE
I am running into this issue in `11.6.0`.  Under a fresh project, it still tries the invalid hash after it is unable to find an existing preference with the valid hash.  This problem goes away if I implement a custom storage handler which bypasses the legacy hash fallback.

### ISSUE THREAD

**Issue by fujohnwang** [dlsc-software-consulting-gmbh/PreferencesFX#286]
I am running into this issue in `11.6.0`.  Under a fresh project, it still tries the invalid hash after it is unable to find an existing preference with the valid hash.  This problem goes away if I implement a custom storage handler which bypasses the legacy hash fallback.

Stack trace:
```
Caused by: java.lang.IllegalArgumentException: Key contains code point U+0000
	at java.prefs/java.util.prefs.AbstractPreferences.get(AbstractPreferences.java:296)
	at com.dlsc.preferencesfx.util.PreferencesBasedStorageHandler.getSerializedPreferencesValue(PreferencesBasedStorageHandler.java:302)
	at com.dlsc.preferencesfx.util.PreferencesBasedStorageHandler.loadObject(PreferencesBasedStorageHandler.java:227)
	at com.dlsc.preferencesfx.model.Setting.loadSettingValue(Setting.java:517)
	at com.dlsc.preferencesfx.model.PreferencesFxModel.lambda$loadSettingValues$7(PreferencesFxModel.java:213)
	at com.dlsc.preferencesfx.model.PreferencesFxModel$$Lambda$199/0000000000000000.accept(Unknown Source)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at com.dlsc.preferencesfx.model.PreferencesFxModel.loadSettingValues(PreferencesFxModel.java:209)
	at com.dlsc.preferencesfx.PreferencesFx.init(PreferencesFx.java:69)
	at com.dlsc.preferencesfx.PreferencesFx.<init>(PreferencesFx.java:64)
	at com.dlsc.preferencesfx.PreferencesFx.<init>(PreferencesFx.java:55)
	at com.dlsc.preferencesfx.PreferencesFx.of(PreferencesFx.java:102)
```

_Originally posted by @incendium in https://github.com/dlsc-software-consulting-gmbh/PreferencesFX/issues/64#issuecomment-583807443_

**Comment by fujohnwang** [comment]
I encounter same problem and had thought it's a problem of JavaPreferences API or input method thing.

**Comment by martinfrancois** [comment]
Should be fixed by 11.7.0, can you please confirm?

**Comment by fujohnwang** [comment]
seems OK now ;)
